core/storage/merkle/polkadot_trie_db/polkadot_codec.cpp

```diff
@@ -189,10 +189,15 @@ namespace kagome::storage::merkle {
     encoding += ushortToBytes(node.childrenBitmap());
 
     // encode each child
-    for (const auto &child : node.children) {
+    for (auto &child : node.children) {
       if (child) {
-        OUTCOME_TRY(encChild, encodeNode(*child));
-        encoding += encChild;
+        if (child->isDummy()) {
```

Comment:
what is dummy node? and where did you get it?

Reply:
Did you check the comments for the dummy node structure? If they are not clear I must rewrite them.